src/main/java/net/minecraftforge/fmp/item/ItemMultipart.java

```diff
@@ -0,0 +1,83 @@
+package net.minecraftforge.fmp.item;
+
+import net.minecraft.block.SoundType;
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.item.Item;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.EnumActionResult;
+import net.minecraft.util.EnumFacing;
+import net.minecraft.util.EnumHand;
+import net.minecraft.util.SoundCategory;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.Vec3d;
+import net.minecraft.world.World;
+import net.minecraftforge.fmp.multipart.IMultipart;
+import net.minecraftforge.fmp.multipart.MultipartHelper;
+
+/**
+ * {@link ItemMultipart} is an {@link Item} that can handle multipart placement.<br/>
+ * Implement {@link ItemMultipart#place(World, BlockPos, EnumFacing, Vec3, ItemStack, EntityPlayer)} and optionally
+ * override {@link ItemMultipart#consumeItem(ItemStack)} and {@link ItemMultipart#getPlacementSound(ItemStack)} to place
+ * your part in the world.
+ */
+public abstract class ItemMultipart extends Item implements IItemMultipartFactory
```

Comment:
That is right, yes. Modders extend it and they have a method which gives them all the information they could possibly need to place their multipart. I was considering the idea of making another version of this class (one that extended it) which would just take a factory as an argument for J8 lambda support, but I'd like feedback on that idea.